src/legacy-html-from-paste.js

```diff
@@ -24,7 +24,15 @@ function getHtmlUsingHiddenElement (hiddenElement) {
   return hiddenElement.innerHTML
 }
 
+// Check for write access to clipboard, otherwise we're not allowed by the browser to paste in a contenteditable container
+function haveClipboardAccess () {
+  return window.clipboardData && window.clipboardData.setData('text/plain', '')
```

Comment:
Do you know if this is replacing data in a users clipboard?

Reply:
Yes it does set `Text` data, but it seems like our `document.execCommand('paste')` doesn't rely on `clipboardData` (I've re-tested to make sure we're not breaking the case where clipboard access is allowed). I can dig a bit more into this to understand what actually happens.

Reply:
I did a bit of 'timeboxed' investigation on this. It turns out the `clipboardData` is wiped out by our function that tests for clipboard access (as expected), but the `document.execCommand('paste')` brings the copied content back to the clipboard (persistently). [MSDN is not very helpful explaining these things](https://msdn.microsoft.com/en-us/ie/ms535220(v=vs.94)) so I have to rely only on manual testing and debugging.

I can keep looking into it, but I'm afraid I will spend hours without a reasonable answer. I am not comfortable pushing code that is not properly explained, but on the other hand I don't see any way to document this without a source or a reference from Microsoft.

Reply:
Sure, that sounds reasonable.

So a few things to test to just try make sure we don't do anything surprising to anyones computer and document in our commit message:

- Basic: We can paste once into our input and the clipboard data is presented
- We can paste twice into our input and the clipboard data is presented both times
- We can paste into our input and then one on another web page and both times the clipboard data is presented
- We can paste into our input and then into another application (say Notepad) and both times the clipboard data is presented

Reply:
Included 1, 2 and 4 into my rounds of tests. Will cover 3 too and update.

Reply:
Works consistent in all scenarios. Tried them all with access to clipboard disabled, enabled and with prompt.